### PR TITLE
Add basic support for interrupt handlers

### PR DESCRIPTION
**Pull request by lbenard**
Basic support for interrupt handlers. The `Store` can create the interrupt handler by itself so I changed the constructor to take an already existing ptr

### CODE REVIEW

File: wasmtime-jni/src/io_github_kawamuray_wasmtime_Store/imp.rs [hunk 1]
```diff
@@ -61,4 +61,10 @@ impl<'a> JniStore<'a> for JniStoreImpl {
         store.gc();
         Ok(())
     }
+
+    fn interrupt_handle_ptr(env: &JNIEnv, this: JObject) -> Result<jlong, Self::Error> {
+        let store = interop::get_inner::<Store<StoreData>>(&env, this)?;
+        let interrupt_handle = store.interrupt_handle()?;
```
Comment (kawamuray) [inline wasmtime-jni/src/io_github_kawamuray_wasmtime_Store/imp.rs:67]: Currently this value never dropped anywhere, which will cause memory leak.
We have to implement `Disposable` to `InterruptHandle` class and implement `dispose()` methods like other classes.

**Review by kawamuray — CHANGES_REQUESTED** [review]
Thanks for the PR! Left few comments.

**Comment by lbenard** [comment]
Should be good, let me know if there's anything else missing!

**Review by kawamuray — APPROVED** [review]
LGTM, thanks!

**Pull request merged**